frontend_tests/casper_tests/04-compose.js

```diff
@@ -117,6 +117,26 @@ casper.waitUntilVisible('#compose', function () {
         'Recipients are displayed correctly in a huddle reply');
 });
 
+
+casper.waitUntilVisible('#markdown_preview', function () {
+    casper.test.assertNotVisible('#undo_markdown_preview', 'Write button is hidden');
+    casper.click("#markdown_preview");
+    casper.test.assertNotVisible('#markdown_preview', 'Preview button is hidden.');
```

Comment:
I think you might want to do this as a waitWhileVisible (and similarly with the test below) so the browser has time to rerender the page; the current code looks like it'll potentially have nondeterministic errors...